### PR TITLE
Plugins: Remove site plugins

### PR DESCRIPTION
**Pull request by rjernst**
Site plugins used to be used for things like kibana and marvel, but
there is no longer a need since kibana (and marvel as a kibana plugin)
uses node.js. This change removes site plugins, as well as the flag for
jvm plugins. Now all plugins are jvm plugins.

### CODE REVIEW

**Comment by s1monw** [comment]
@rjernst this LGTM I guess plugins that used this behavior will be able to just register handlers and serve static content from there. I wonder if we should add an example somewhere how to do that? this must not block this issue in any case I would just want to show folks that used that how to proceed from here. WDYT? Don't get me wrong it's not on you to do that I just wanna get your opinion

**Comment by rjernst** [comment]
@s1monw Is that what we really recommend? Site plugins were for using ES as a generic webserver...I'm not sure we should continue that? It seems like suggesting to build kibana plugins or serve the files with a real webserver is better?  Site plugins just served static files right?

**Comment by s1monw** [comment]
> @s1monw Is that what we really recommend? Site plugins were for using ES as a generic webserver...I'm not sure we should continue that? It seems like suggesting to build kibana plugins or serve the files with a real webserver is better? Site plugins just served static files right?

I guess that's a better alternative... lets see how we can communicate that best

**Comment by clintongormley** [comment]
I've chatted to the Kibana guys about how easy it is to migrate a site plugin to Kibana and the answer is: pretty easy.  There is a Yeoman generator for Kibana plugins (https://github.com/elastic/generator-kibana-plugin) which generates all the boiler plate you need, and the rest is pretty much copy and paste.  There are one or two things to watch out for (eg using the same version of jQuery for instance), but the rest should be pretty easy.  We will do a blog post demonstrating how to migrate a site plugin to Kibana as a conversion guide for our community authors.

Once migrated, these kibana plugins can start taking advantage of some of the functionality available in Kibana, although I think most of that will probably come later - the public API is still in flux.

Site plugins have always been a bit of a hack, and have been the source of a few CVEs in Elasticsearch's history, so it'll be good to reduce the attack surface area here.  Also, the fact that Kibana is backed by a node.js server opens up more possibilities to plugin authors.

**Comment by clintongormley** [comment]
@rjernst could we have a note in the breaking changes docs for 3.0 please

**Comment by rjernst** [comment]
Added a note: 60180fe

**Comment by jettro** [comment]
When migrating site plugins that do more than just a few queries the migration is more complicated. A best practise of interacting with the connected elasticsearch cluster would be cool. I see Kibana uses a proxy with very limited functionality. Sense does something else to connect to elastic. What is the path to chose for you migrated site plugin?

**Comment by clintongormley** [comment]
@jettro Sense sends its requests via the Kibana backend.  One advantage to this is that you can have multiple Elasticsearch clusters setup in Kibana and just have the one Kibana instance to talk to them.  You don't then need to worry about configuring CORS settings on each cluster.  We will be following up this PR with a blog post demonstrating how to migrate, which will include using the Kibana backend as a proxy.

**Comment by javanna** [comment]
++ on the blogpost and thanks @jettro for bringing this up ;)

File: plugins/site-example/build.gradle [hunk 1]
```diff
@@ -1,27 +0,0 @@
-/*
```
Comment (dadoonet) [inline plugins/site-example/build.gradle]: @rjernst there is a link in the documentation to this file: https://www.elastic.co/guide/en/elasticsearch/plugins/master/plugin-authors.html
Is that something you could fix?

Comment (rjernst) [inline plugins/site-example/build.gradle]: Not sure what you mean. That page is already up to date (it does not reference site plugins).

Comment (dadoonet) [inline plugins/site-example/build.gradle]: I meant that in this doc: https://raw.githubusercontent.com/elastic/elasticsearch/master/docs/plugins/authors.asciidoc

There is a link to `https://github.com/elastic/elasticsearch/blob/master/plugins/site-example/build.gradle` which has been removed with this PR.

Comment (rjernst) [inline plugins/site-example/build.gradle]: Ok, I missed it, thanks. I remove it in 3f44e1d

**Comment by rashidkpc** [comment]
I still think we should talk about deprecating this instead of removing it. 

The Kibana plugin API, like the Elasticsearch plugin API, is not stable. If we expect site plugin authors to regularly update we're going to be in trouble, kopf hasn't had a commit in 2 months, head hasn't had one in a month. I don't think this is a good idea.

**Comment by rashidkpc** [comment]
Also, we should probably pull the list of site plugins out of the docs: https://github.com/elastic/elasticsearch/search?utf8=%E2%9C%93&q=kopf

**Pull request merged**